code/modules/fishing/fishing_gear.dm

```diff
@@ -350,3 +350,341 @@ TYPEINFO(/obj/item/fish_portal)
+			user.changeStatus("weakened", 5 SECONDS)
+			user.TakeDamage(user.hand == LEFT_HAND ? "l_arm": "r_arm", 15, 0, 0, DAMAGE_STAB)
+			user.force_laydown_standup()
+			implanted(user, null)
+		else
+			return ..()
+
+	throw_impact(mob/hit_atom, datum/thrown_thing/thr)
+		if (isliving(hit_atom) && hit_atom.equipped() == src.rod)
+			return
+		else if ((isliving(hit_atom) && !hit_atom.nodamage && GET_DIST(src, src.rod) < src.rod.line_length))
+			implanted(hit_atom, null)
+		return ..()
+
+	implanted(mob/M, mob/I)
+		if (istype(M))
+			M.visible_message("<span class='alert'><b>[M] gets snagged by a fishing lure!</b></span>")
+			logTheThing(LOG_COMBAT, M, "is caught by a barbed fishing lure at [log_loc(src)]")
+			M.emote("scream")
+			take_bleeding_damage(M, null, 10, DAMAGE_STAB)
+			M.UpdateDamageIcon()
+		return ..()
+
+	on_life(mult)
+		. = ..()
+		if (GET_DIST(src, src.rod) > src.rod.line_length && src.owner)
+			src.owner.visible_message("\The [src] rips out of [src.owner]!", "\The [src] rips out of you!")
+			take_bleeding_damage(src.owner, null, 5, DAMAGE_STAB)
+			src.set_loc(get_turf(src.loc))
+			src.on_remove(src.owner)
+
+	Eat(mob/M, mob/user, by_matter_eater)
+		. = ..()
+		M.emote("scream")
+		M.TakeDamage("chest", 25, 0, 0, DAMAGE_CUT)
+		M.visible_message("\The [src] tears a bunch of gore out of [M.name]!")
+		if (M.bioHolder && M.bioHolder.Uid && M.bioHolder.bloodType)
+			gibs(M.loc, blood_DNA=M.bioHolder.Uid, blood_type=M.bioHolder.bloodType, headbits=FALSE, source=M)
+		else
+			gibs(M.loc, headbits=FALSE, source=M)
+		var/mob/living/carbon/human/H = M
+		if (istype(H))
+			if (H.organHolder)
+				for(var/organ in list("right_kidney", "left_kidney", "liver", "stomach", "intestines", "spleen", "pancreas"))
+					var/obj/item/organ/O = H.drop_organ(organ, M.loc)
+					if (istype(O))
+						O.throw_at(src.rod.loc, rand(3,6), rand(1,2))
+		qdel(src)
+
+	disposing()
+		. = ..()
+		src.rod.lure = null
+
+//action (with bar) for reeling in a mob with the Glaucus
+/datum/action/bar/syndie_fishing
+	interrupt_flags = INTERRUPT_STUNNED | INTERRUPT_ACTION
+	var/mob/user = null
+	var/mob/target = null
+	/// what fishing rod caught the mob
+	var/obj/item/syndie_fishing_rod/rod = null
+	/// what lure is snagged in the mob
+	var/obj/item/implant/syndie_lure/lure = null
+	/// stores current damage per point blank reel and increases by 3 each cycle that target isnt point blank
+	/// resets when damage is dealt
+	var/damage_on_reel = 7
+	/// how long a step of reeling takes, set onStart
+	duration = 0
+	/// id for fishing action
+	id = "fishing_for_fools"
+
+	New(var/user, var/target, var/rod, var/lure)
+		..()
+		src.user = user
+		src.target = target
+		src.rod = rod
+		src.lure = lure
+
+	onStart()
+		..()
+		if (!src.user || !src.target || !src.rod || !src.lure || (src.target == src.user) || !(src.lure.owner == src.target) || !(src.user.equipped() == src.rod) || !isturf(src.user.loc) || GET_DIST(src.user, src.target) > src.rod.line_length)
```

Comment:
please make this a proc or a local macro or something, that's a long if chain to be duplicating three times

Reply:
Or refactor to use `canRunCheck()`